src/checkers/inference/solver/backend/SolverType.java

```diff
@@ -0,0 +1,79 @@
+package checkers.inference.solver.backend;
+
+import java.lang.reflect.Constructor;
+import java.util.Collection;
+import java.util.Map;
+
+import javax.annotation.processing.ProcessingEnvironment;
+
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.javacutil.ErrorReporter;
+
+import checkers.inference.model.Constraint;
+import checkers.inference.model.Serializer;
+import checkers.inference.model.Slot;
+import checkers.inference.solver.backend.logiql.LogiQLFormatTranslator;
+import checkers.inference.solver.backend.logiql.LogiQLSolver;
+import checkers.inference.solver.backend.maxsat.LingelingSolver;
+import checkers.inference.solver.backend.maxsat.MaxSatFormatTranslator;
+import checkers.inference.solver.backend.maxsat.MaxSatSolver;
+import checkers.inference.solver.frontend.Lattice;
+
+public enum SolverType {
+
+    MAXSAT("MaxSAT", MaxSatSolver.class, MaxSatFormatTranslator.class),
+    LINGELING("Lingeling", LingelingSolver.class, MaxSatFormatTranslator.class),
+    LOGIQL("LogiQL", LogiQLSolver.class, LogiQLFormatTranslator.class);
+
```

Comment:
Z3 required here?

Reply:
I will add Z3 by merging #71 . So I just separate Z3 implementation from this PR.

Reply:
Sounds good